### PR TITLE
Fix of broken API Auth

### PR DESCRIPTION
**Pull request by guillim**
Fix done this morning with @FelixMalfait  from #8295

### CODE REVIEW

**Review by FelixMalfait — APPROVED** [review]
Great first PR! 🙌

File: packages/twenty-server/src/engine/core-modules/auth/strategies/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { WorkspaceTokenType } from 'src/engine/core-modules/jwt/services/jwt-wrapper.service';
+
+export type JwtPayload = {
+  sub: string;
+  workspaceId?: string;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/auth/strategies/interfaces.ts:5]: should be required

File: packages/twenty-server/src/engine/core-modules/auth/strategies/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { WorkspaceTokenType } from 'src/engine/core-modules/jwt/services/jwt-wrapper.service';
+
+export type JwtPayload = {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/auth/strategies/interfaces.ts:3]: types/JwtPayload

File: packages/twenty-server/src/engine/core-modules/auth/strategies/jwt.auth.strategy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  workspaceRepository = {
+    findOneBy: jest.fn(async () => new Workspace()),
+  };
+
+  userRepository = {
+    findOne: jest.fn(async () => null),
+  };
+
+  // dataSourceService = {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/auth/strategies/jwt.auth.strategy.spec.ts:30]: comments?

File: packages/twenty-server/src/engine/core-modules/environment/environment-variables.ts [hunk 4]
```diff
@@ -137,6 +137,11 @@ export class EnvironmentVariables {
   @IsString()
   APP_SECRET: string;
 
+  // legacy
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/environment/environment-variables.ts:140]: TODO: Legacy to be removed

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts [hunk 5]
```diff
@@ -58,6 +59,16 @@ export class JwtWrapperService {
     }
 
     try {
+      // we check that ACCESS_TOKEN_SECRET is set
+      const appSecret = this.environmentService.get('ACCESS_TOKEN_SECRET');
+
+      if (!type && !payload.workspaceId && appSecret) {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts:65]: remove appSecret

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts [hunk 6]
```diff
@@ -58,6 +59,16 @@ export class JwtWrapperService {
     }
 
     try {
+      // we check that ACCESS_TOKEN_SECRET is set
+      const appSecret = this.environmentService.get('ACCESS_TOKEN_SECRET');
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts:63]: accessTokenSecret

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts [hunk 7]
```diff
@@ -93,4 +104,19 @@ export class JwtWrapperService {
       .update(`${appSecret}${workspaceId}${type}`)
       .digest('hex');
   }
+
+  generateAppSecretLegacy(
+    type: WorkspaceTokenType,
+    workspaceId?: string,
+  ): string {
+    const appSecret = this.environmentService.get('ACCESS_TOKEN_SECRET');
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/jwt/services/jwt-wrapper.service.ts:112]: accessTokenSecret

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**